arangod/Aql/Ast.h

```diff
@@ -474,6 +474,8 @@ class Ast {
   /// @brief determines the variables referenced in an expression
   static void getReferencedVariables(AstNode const*, VarSet&);
 
+  static bool intersectsWith(AstNode const*, VarSet const&);
```

Comment:
Better would be name it like "isVarsUsed" or smth.  As "intersection" between `AstNode` and `VarSet`  is not a very clear concept.